dotnetv3/cross-service/S3ObjectLambdaFunction/README.md

```diff
@@ -0,0 +1,204 @@
+
+4. Choose Empty Function Blueprint
+
+![step-4](images/step-4.png)
+
+5. Install Nuget Package - AWSSDK.S3 & Amazon.Lambda.S3Events
+
+![step-5](images/step-5.png)  
+![step-6](images/step-6.png) 
+
+6. Update the Function Code
+
+```csharp
+public class Function
+{
+    IAmazonS3 S3Client { get; set; }
+
+    public Function()
+    {
+        S3Client = new AmazonS3Client();
+    }
+    public Function(IAmazonS3 s3Client)
+    {
+        this.S3Client = s3Client;
+    }
+
+    public async Task<HttpResponseMessage> FunctionHandler(S3ObjectLambdaEvent s3Event, ILambdaContext context)
+    {
+        try
+        {
+            var objectContext = s3Event.GetObjectContext;
+            var s3Url = objectContext.InputS3Url;
+            var requestRoute = objectContext.OutputRoute;
+            var requestToken = objectContext.OutputToken;
+            context.Logger.LogInformation(objectContext.InputS3Url.ToString());
+
+            var httpClient = new HttpClient();
+
+            var response = await httpClient.GetAsync(s3Url);
+            var responseBody = await response.Content.ReadAsStringAsync();
+            var transformedObject = responseBody.ToUpper();
+
+            var writeGetObjectResponseRequest = new Amazon.S3.Model.WriteGetObjectResponseRequest
+            {
+                Body = GenerateStreamFromString(transformedObject),
+                RequestRoute = requestRoute,
+                RequestToken = requestToken
+            };
+
+            await S3Client.WriteGetObjectResponseAsync(writeGetObjectResponseRequest);
+
+            return new HttpResponseMessage(System.Net.HttpStatusCode.OK);
+        }
+        catch (Exception e)
+        {
+            context.Logger.LogError(e.Message);
+            context.Logger.LogError(e.StackTrace);
+            throw;
+        }
+    }
+
+    public static MemoryStream GenerateStreamFromString(string value)
+    {
+        return new MemoryStream(Encoding.UTF8.GetBytes(value ?? ""));
+    }
+}
+```
+
+7. Add Using statement for Amazon.Lambda.S3Event
+
+```csharp
+using Amazon.Lambda.S3Events;
+```
+
+8. Right Click the Project in the Solution Explorer and choose "Publish to AWS Lambda".
+![step-7](images/step-7.png)
+
+9. Choose Create a new function and give a name to the Lambda function and click Next.
+![step-8](images/step-8.png)
+
+10. Choose Role as New Role based on AWSLambdaBasicExecutionRole and click upload.
+![step-9](images/step-9.png)
+
+11. Validate Lambda in AWS Console.
+![step-10](images/step-10.png)
+
+12. Update the Permission, Navigate to the Permission and go to IAM by clicking the Role name Link.
+![step-11](images/step-11.png)
+
+13. Update the Permission with "AmazonS3ObjectLambdaExecutionRolePolicy".
+![step-12](images/step-12.png)
+
+
+## Create an S3 Object Lambda Access Point
+
+Refer to [Create an S3 Object Lambda Access Point](https://docs.aws.amazon.com/AmazonS3/latest/userguide/tutorial-s3-object-lambda-uppercase.html#ol-upper-step6)
```

Comment:
The previous section includes a text step for each image. I think this section could use that as well, because all of the images together without text may be confusing.

Reply:
Updated